### ISSUE TITLE
[BUG] `temporal_train_test_split` does not work on panel datatypes with unequal length series.

### ISSUE THREAD

**Issue by hazrulakmal** [sktime/sktime#5107]
**Describe the bug**

Also relates to #4968 

`temporal_train_test_split` wrongly split panel datatypes (splitting per unique unequal  time series). It could be that the split function does not support this type yet, If so, it should throw an error msg telling that it does not currently supported the datatypes.

**To Reproduce**

```python
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.utils._testing.panel import _make_panel

y = _make_panel(n_instances=2, n_timepoints=6)
# make first instance series shorter than the second
y.iloc[4:6] =None
y.dropna(inplace=True)

train_size, test_size = temporal_train_test_split(y,  test_size=2) 
# show shapes
print(
    f"""
    {y.shape=} 
    {train_size.shape=}, {test_size.shape=} #train size should be (6,1) 2+4=6
    """
)

# has also the same issue as #4968 below is the minimal example
train_fh, test_fh = temporal_train_test_split(y,  fh=[1,2]) 
# show shapes
print(
    f"""
    {y.shape=} 
    {train_fh.shape=}, {test_fh.shape=} #train size should be (6,1) and test (4,1)
   """
)
```

output
```
y.shape=(10, 1) 
train_size.shape=(4, 1), test_size.shape=(4, 1) #train size should be 6 (2+4)

y.shape=(10, 1) 
train_fh.shape=(8, 1), test_fh.shape=(2, 1) #train size should be (6,1) and test (4,1)
```

**Versions**
<details>

 main at [3cf69ed](https://github.com/sktime/sktime/commit/3cf69eddba315d6130b661ca5fe8e132e236aa47)

</details>

**Comment by fkiraly** [comment]
In my view, the reason is that `temporal_train_test_split` is, under the hood, a splitter, but it is a completely separate interface and does not inherit boilerplate from `BaseSplitter` which would give it broadcasting over panels/hierarchical etc.

I think the one thing to do here is to turn the parametrization under `temporal_train_test_split` into a splitter, and call that new splitter's `split_series` method under the hood.

**Comment by fkiraly** [comment]
I see that in the bug report #5108 you used `ExpandingGreedySplitter` to replace `temporal_train_test_split` - perhaps that's a good starting point?

(FYI @dabidgilbertson)

**Comment by hazrulakmal** [comment]
> ExpandingGreedySplitter to replace temporal_train_test_split - perhaps that's a good starting point?
yup, it's a good starting point. but only for a case when the test set is at the end of a given input y.

The `temporal_test_split` function supports an 'fh' argument, which enables users to specify a custom test date. This date can be set to any time between the start and end dates of the input y. If a user wants, say, the test set to be in the middle of y, the `ExpandingGreedySplitter` cannot handle this (is probably not designed to do so).

so I think your idea of having a `TrainTestSplitter` is perhaps the solution to handle shortfalls in `temporal_train_test_split` and is good for maintenance too.

**Comment by yarnabrina** [comment]
> If a user wants, say, the test set to be in the middle of y

Just for the sake of argument, is it realistic for forecasting, if user uses just train+test folds?

**Comment by JonathanBechtel** [comment]
@yarnabrina It's not super common, but I know of some validation methods that call for that.  There is a temporal CV method called purged-combinatorial-kfold which does exactly that.  The idea is that if you "embargo" the middle portion with enough distance between the beginning and middle parts then you can ensure you're not leaking any temporal information to the validation set.  

It's pretty niche, but it's out there.

**Comment by hazrulakmal** [comment]
I may have missed this. but what I meant for clarity is as follows

```python
 | * * * * * x x x - - - |
```

where `x` is training time points, `x` is test time points and `-` is time points you discard.

you can do this with `temporal_train_test_split` but not `ExpandingGreedySplitter`. Essentially the output behaviours are different in this sense.

**Comment by fkiraly** [comment]
can you kindly make precise the parameters in `temporal_train_test_split` that give rise to the picture?

**Comment by hazrulakmal** [comment]
```python
import pandas as pd

from sktime.forecasting.base import ForecastingHorizon
from sktime.split import temporal_train_test_split
from sktime.datasets import load_longley

y, _ = load_longley()  # has a yearly period index from 1947 to 1962
date_range = pd.period_range(start="1955", end="1958", freq="Y")
fh = ForecastingHorizon(date_range, is_relative=False)  # or fh = date_range
train, test = temporal_train_test_split(y, fh=fh)

test.index
```

the test index output is `PeriodIndex(['1955', '1956', '1957', '1958'], dtype='period[A-DEC]')`

**Comment by fkiraly** [comment]
I see. That's pretty odd, but it is an example.
